find_cmake_package_name.py

```diff
@@ -78,40 +74,91 @@ def find_path(conan_root, package_info):
     """
 
     package_name, version, repository, branch = package_info
-    package_path = "{}/{}/{}/{}/{}".format(conan_root, package_name, version,
-                                           repository, branch)
+    package_path = f"{conan_root}/{package_name}/{version}/{repository}/{branch}"
 
     if not path.isdir(package_path):
         print(
-            "Error: path name \"{}\" is not a directory.".format(package_path),
+            'Error: path name "{package_path}" is not a directory.',
             file=sys.stderr)
         exit(1)
     return package_path
 
 
-def find_configurations(package_path):
+config_suffix = re.compile('(C|-c)onfig.cmake$')
+
+
+def find_configuration(package_path):
     """Find CMake configuration files.
 
     package_path -- The directory path to start searching from.
     """
 
-    config_suffix = re.compile('(C|-c)onfig.cmake$')
-    f = list()
-    for _, _, files in walk(package_path):
-        f += list(filter(lambda file: config_suffix.search(file), files))
-    f = list(dict.fromkeys(f))
-    return list(map(lambda i: config_suffix.sub('', i), f))
+    configurations = list()
 
+    for root, _, files in walk(package_path):
+        configurations += list(
+            filter(lambda file: config_suffix.search(file), files))
 
-def main():
-    package_ref, conan_root = parse_args()
-    package_path = find_path(conan_root, split_package_ref(package_ref))
+    configurations = list(
+        map(lambda i: config_suffix.sub('', i), dict.fromkeys(configurations)))
+
+    if len(configurations) == 0:
+        raise RuntimeError("Could not find configuration")
 
-    configurations = find_configurations(package_path)
+    return configurations[0]
+
+
+target_suffix = re.compile('Targets.cmake$')
+
+
+def find_linkage(package_path, package_name):
+    """Find CMake configuration files.
 
-    for i in configurations:
-        print(i)
-    exit(len(configurations) == 0)
+    package_path -- The directory path to start searching from.
+    package_name -- The name of the package
+    """
+
+    for root, _, files in walk(package_path):
+        for cmake_file in filter(
+                lambda f: config_suffix.search(f) or target_suffix.search(f),
+                files):
+            with open(f"{root}/{cmake_file}") as to_read:
+                data = to_read.read()
+                if "::" in data:
+                    result = re.search("([A-Za-z0-9_-]+)::[A-Za-z0-9_-]+",
+                                       data)
+                    if result[1] == package_name:
+                        return result[0]
+    return package_name
+
+
+def print_template(path, substitutions):
+    """
+        A crude way to handle string templating until I work out string template engines in Python.
```

Comment:
I personally like using this for small things: https://github.com/defunkt/pystache
If you need something that does more than substitution: http://jinja.pocoo.org/docs/2.10/

Reply:
Thanks! I ended up using `pystache`.